misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -87,6 +88,14 @@ protected JsonElement transformSyntaxNode(Node node) {
         }
         nodeJson.addProperty("source", node.toSourceCode());
         nodeJson.addProperty("kind", prettifyKind(node.kind().toString()));
+        // TODO: Generalize the diagnostic implementation.
+        Iterable<Diagnostic> syntaxDiagnostics = node.diagnostics();
+        if (syntaxDiagnostics != null) {
+            nodeJson.add("syntaxDiagnostics", DiagnosticUtil.getDiagnostics(syntaxDiagnostics));
+        }
+        JsonArray responseMinutiae = evaluateLeadingMinutiae(node.leadingMinutiae());
+        nodeJson.add("leadingMinutiae", responseMinutiae.get(1));
+        nodeJson.add("trailingMinutiae", assignMinutiae(node.trailingMinutiae()));
```

Comment:
Is there specific reason to treat the leading and trailing minutiae differently? Isn't the approach of using `evaluateLeadingMinutiae` still valid for trailing minutiae?